common/src/View/MapDocumentCommandFacade.cpp

```diff
@@ -903,8 +904,16 @@ namespace TrenchBroom {
             const auto& faceHandles = selectedBrushFaces();
             if (!faceHandles.empty()) {
                 snapshot->restoreNodes(m_worldBounds);
-                
-                setTextures(faceHandles);
+
+                // Restoring the snapshots will invalidate all texture pointers on the BrushNode,
+                // since the snapshot has a whole brush granularity, so we need to call
+                // setTextures on the whole node.
+                kdl::vector_set<Model::Node*> nodes;
```

Comment:
Maybe pass the number of the face handles as the capacity to avoid reallocations.

Reply:
Yeah good idea